### PR TITLE
Add: Added edit feature

### PR DESCRIPTION
**Pull request by zoef-yt**
In this pr, I added

- [ ] Edit feature on an existing card. Completing the p0 for components
- [ ] Subtle animations on the card

### CODE REVIEW

**Review by humzahasan — APPROVED** [review]
Code changes look Good.
Liked the animation you added

Good to merge!

**Review by rutu14 — APPROVED** [review]
The code is good to merge.
UI is amazing, the animation is seamless.

File: src/Components/NotesCard/NotesCard.css [hunk 1]
```diff
@@ -4,6 +4,50 @@
 	max-width: 40rem;
 	height: fit-content;
 	max-height: 40rem;
+	animation-name: notes-card-animation;
+	animation-duration: 0.4s;
+}
+
+@keyframes notes-card-animation {
```
Comment (muskanbagrecha) [inline src/Components/NotesCard/NotesCard.css:11]: This animation is so subtle yet looks so amazing. Great stuff.

File: src/Components/NotesCard/NotesCard.jsx [hunk 2]
```diff
@@ -33,25 +34,39 @@ const NotesCard = ({ note, isHomePage = false, isArchive = false, isTrash = fals
...
-				<p>Priority:{priority}</p>
-				{tag !== 'None' && <p>Tag:{tag}</p>}
+			<div className='flex-row'>
+				<h3 title={title} className='notes-card-title'>
+					{title}
+				</h3>
+				<div className='notes-card-tags'>{tag !== 'None' && `#${tag}`}</div>
 			</div>
 			<hr />
 			<div style={{ width: '100%' }} className=' rdw-editor-main' dangerouslySetInnerHTML={{ __html: body }}></div>
```
Comment (muskanbagrecha) [inline src/Components/NotesCard/NotesCard.jsx]: Avoid inline styling.

File: src/Components/NotesCard/NotesCard.jsx [hunk 3]
```diff
@@ -33,25 +34,39 @@ const NotesCard = ({ note, isHomePage = false, isArchive = false, isTrash = fals
...
+						<button
+							title='Send to trash page'
+							onClick={() => notesToTrashHandler(_id, note)}
+							className='btn btn-secondary notes-card-btn'
+						>
 							<TrashIcon />
 						</button>
 
 						<button
+							title='Send to archive page'
```
Comment (muskanbagrecha) [inline src/Components/NotesCard/NotesCard.jsx:69]: Good you're providing title as some might misunderstand the icons.

File: src/Components/NotesPage/NotesPage.jsx [hunk 4]
```diff
@@ -86,13 +102,14 @@ const NotesPage = () => {
 							tagChangeHandler={tagChangeHandler}
 							priorityChangeHandler={priorityChangeHandler}
 							colorChangeHandler={colorChangeHandler}
+							updateNoteHandler={updateNoteHandler}
 						/>
 					</Modal>
 
 					{notes.length > 0 ? (
 						getFilterData(filterState).length > 0 ? (
```
Comment (muskanbagrecha) [inline src/Components/NotesPage/NotesPage.jsx]: You can call the function only once and store the result in a variable.

File: src/Components/NotesPage/modal/AddNewNoteCard.jsx [hunk 5]
```diff
@@ -14,8 +14,9 @@ function AddNewNoteCard({
 	tagChangeHandler,
 	priorityChangeHandler,
 	colorChangeHandler,
+	updateNoteHandler,
 }) {
-	const { title, body, tag, priority, cardColor, createdAt } = formData;
+	const { title, tag, priority, cardColor, isEdit, _id } = formData;
 	const isBodyEmpty = editorState.getCurrentContent().getPlainText().length === 0;
```
Comment (muskanbagrecha) [inline src/Components/NotesPage/modal/AddNewNoteCard.jsx]: image
Please use trim function.

File: src/Components/NotesPage/modal/AddNewNoteCard.jsx [hunk 6]
```diff
@@ -57,9 +58,9 @@ function AddNewNoteCard({
 				<button
 					title={`${canSubmit ? 'Add to notes' : 'Cannot leave the fields blank'}`}
 					className={`btn ${canSubmit ? 'btn-primary' : 'btn-disabled'} `}
-					onClick={canSubmit ? () => submitNote() : null}
+					onClick={canSubmit ? (isEdit ? () => updateNoteHandler(_id, formData) : () => submitNote()) : null}
```
Comment (muskanbagrecha) [inline src/Components/NotesPage/modal/AddNewNoteCard.jsx:61]: This is slightly confusing. Extract it out.

**Review by muskanbagrecha — CHANGES_REQUESTED** [review]
Great work. The final result looks great. Clean and easy-to-understand code. Can look into suggested changes.

File: src/Components/NotesCard/NotesCard.jsx (hunk 2 above)
Comment (zoef-yt) [reply src/Components/NotesCard/NotesCard.jsx]: Done

File: src/Components/NotesPage/NotesPage.jsx (hunk 4 above)
Comment (zoef-yt) [reply src/Components/NotesPage/NotesPage.jsx]: Done.

File: src/Components/NotesPage/modal/AddNewNoteCard.jsx (hunk 5 above)
Comment (zoef-yt) [reply src/Components/NotesPage/modal/AddNewNoteCard.jsx]: Done thanks.

File: src/Components/NotesPage/modal/AddNewNoteCard.jsx (hunk 6 above)
Comment (zoef-yt) [reply src/Components/NotesPage/modal/AddNewNoteCard.jsx:61]: Will extract it after the submmision

**Review by muskanbagrecha — APPROVED** [review]
Good to merge as you have implemented the suggested changes. Great work overall, the code looks clean, and the functionalities have come out really well.

**Pull request merged**